### ISSUE TITLE
[BUG] Callback called multiple times

### ISSUE THREAD

**Issue by olegario96** [C2FO/fast-csv#423]
**Describe the bug**
I'm currently using your lib to parsing a stream received from a S3 bucket. But after some time I've noticed the following error:
```JSON
{
    "errorType": "Error",
    "errorMessage": "Callback called multiple times",
    "code": "ERR_MULTIPLE_CALLBACK",
    "stack": ["Error [ERR_MULTIPLE_CALLBACK]: Callback called multiple times", "    at CsvParserStream.afterTransform (_stream_transform.js:89:31)", "    at callNext (/var/task/node_modules/@fast-csv/parse/build/src/CsvParserStream.js:93:28)", "    at /var/task/node_modules/@fast-csv/parse/build/src/CsvParserStream.js:119:28", "    at CsvParserStream.transformRow (/var/task/node_modules/@fast-csv/parse/build/src/CsvParserStream.js:165:13)", "    at iterate (/var/task/node_modules/@fast-csv/parse/build/src/CsvParserStream.js:116:25)", "    at Immediate._onImmediate (/var/task/node_modules/@fast-csv/parse/build/src/CsvParserStream.js:97:40)", "    at processImmediate (internal/timers.js:456:21)", "    at process.topLevelDomainCallback (domain.js:137:15)"]
}
```

**Parsing or Formatting?**

 - [ ] Formatting
 - [X] Parsing 

**To Reproduce**
Here's the current coding that's triggering this error:
```typescript
return new Promise((resolve): void => {
  stream
    .pipe(parseCsv())
    .on('data', async (row: string[]) => {
      buffer.push(row);
      if (buffer.length === this.bufferLimit) {
        const bufferToBeConsumed = buffer;
        buffer = [];
        await this.upsertCsvRowsByExportType(bufferToBeConsumed, filename);
      }
    })
    .once('end', async () => {
      if (buffer.length) {
        const bufferToBeConsumed = buffer;
        buffer = [];
        await this.upsertCsvRowsByExportType(bufferToBeConsumed, filename);
        resolve();
      }
    });
});
```
The lib isn't used in any other place.

**Expected behavior**
No error should occur during parsing process.

**Screenshots**
<img width="832" alt="Screen Shot 2020-07-08 at 13 37 48" src="https://user-images.githubusercontent.com/15036437/86946487-d121f800-c120-11ea-989c-59181c5bdd9d.png">

**Desktop (please complete the following information):**
 - Lambda function
 - Node Version 12.13
 - Lib Version 4.3.1

**Additional context**

**Comment by olegario96** [comment]
Fixed after remove this lib from my dependencies and use `csv-parse` :)

**Comment by rhanka** [comment]
I have the same "Callback called multiple times" each time I try to have the fast-csv parse stream pipe into another one... is this a fast-csv (new?) bug ?

**Comment by doug-martin** [comment]
@rhanka would you mind testing out `v4.3.0` to see if the bug exists in that version? I'm wondering if its related to this commit https://github.com/C2FO/fast-csv/commit/84ecdf6ed18b15d68b4ed3e2bfec7eb41b438ad8

**Comment by rhanka** [comment]
The same occurs with v4.3.1, v4.3.0, v4.2.0 and v4.1.6

**Comment by doug-martin** [comment]
@rhanka thank you for checking. These types of issues are sometimes hard to reproduce, I'll investigate shortly.

**Comment by Roosh513** [comment]
I've identified the problem on this line:

https://github.com/C2FO/fast-csv/blob/23bf809cdee1f5bebd87d0304a4a5b4000ba81c3/packages/parse/src/CsvParserStream.ts#L85

It occurs in both _flush() and _transform(). The processRows() -> callNext() function calls the done() callback, and then is caught by _flush() and _transform(), where the done() callback is invoked again. The error was reproduced by parsing a CSV file containing duplicate headers.

**Comment by Roosh513** [comment]
It looks like this may only occur when running from certain versions of Node. The following block will result in the reported error in Node v8.15.0. It does not cause a problem when running under the currently configured v14.8.0.

```javascript
const CSV_CONTENT = ['a,b,b,c', '1,2,2,3'].join(EOL,);
csv.parseString(CSV_CONTENT, {headers: true})
```

**Comment by chrisnorwood** [comment]
I've been receiving this same error while trying to parse a streamed csv file.

Node Version: 10.18.1
@fast-csv/parse: 4.3.1

**Comment by doug-martin** [comment]
Just issue #464 to try to address this, instead of calling done again in the catch block it will now re throw the error. 

I'll update once I publish

**Comment by doug-martin** [comment]
Just published `v4.3.2`, I'll leave this issue open for a couple of days in case it does not completely address it. 

Thank you for the bug report!
